File: redsheet-portal/src/app/database-models/application-table.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class ApplicationTable {
    constructor(applicationTable?: ApplicationTable )  {
        let properties = [ "applicationTableId", "applicationTableName", "tableTypeId",];
        for (let property of properties)
            if (applicationTable && applicationTable[property])
                this[property] = applicationTable[property];
    }
 
	applicationTableId : number =   0 ;
 
    @required()
    @maxLength(100)
	applicationTableName : string =   undefined;
 
    @range(1,2147483647)
	tableTypeId : number =   undefined;


}

File: RedSheet.DbEntities/Models/ApplicationTable.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ApplicationTables", Schema = "dbo")]
	[Serializable]
	public partial class ApplicationTable  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ApplicationTableId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string ApplicationTableName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TableTypeId { get; set; } 

	public ApplicationTable()
	{
    }

    }
}

